server/lobbyconnection.py

```diff
@@ -113,8 +118,14 @@ def get_user_identifier(self) -> str:
     async def on_connection_made(self, protocol: Protocol, peername: Address):
         self.protocol = protocol
         self.peer_address = peername
+        asyncio.create_task(self.timeout_login())
         metrics.server_connections.inc()
 
+    async def timeout_login(self):
+        await asyncio.sleep(config.LOGIN_TIMEOUT)
+        if not self._authenticated:
+            await self.abort("Client took too long to log in.")
```

Comment:
I did notice that it's possible for this message to be printed if the client already disconnected before the timeout ended. Probably want some cancellation logic here.

Reply:
Added task cancellation when the client disconnects.